src/linalg/impl_linalg.rs

```diff
@@ -6,28 +6,31 @@
 // option. This file may not be copied, modified, or distributed
 // except according to those terms.
 
-use crate::imp_prelude::*;
-use crate::numeric_util;
 #[cfg(feature = "blas")]
 use crate::dimension::offset_from_low_addr_ptr_to_logical_ptr;
+use crate::imp_prelude::*;
```

Comment:
(We don't need to change anything, just talking). It's just weird that tools would move down the implementation prelude import. They are wrong, that import should clearly be first. (Probably accomplished by having it with a blank line after).

Reply:
Personally, I have always liked that Rustfmt sort use statements. Whether sorting them alphabetically is good or a bad is a reasonable discussion to have, but as usual with aesthetics, just the fact that the decision is automatic seems to bring almost all of the benefit, i.e. no more tedious discussions on how to sort here or there or writing up rules as a company-wide guideline.

Additionally, I find the ecosystem-wide readability resulting from the common application of Rustfmt's defaults highly beneficial to upstream collaboration.

(Especially since the workaround using a blank line really seems like a good idea if the import actually is special among the others next to it.)

Reply:
Yeah I agree, it is a bit strange. Perhaps so other imports don't silently clobber items brought in from the prelude?

Reply:
Sounds good @adamreichold. You'll notice that I'm not yet very fond of rustfmt, unfortunately. I'd be happier if it could accept "multiple good ways" to format the same code.